lib/modules/versioning/aws-eks-addon/index.ts

```diff
@@ -0,0 +1,60 @@
+import { regEx } from '../../../util/regex';
+import { coerceString } from '../../../util/string';
+import type { GenericVersion } from '../generic';
+import { GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'aws-eks-addon';
+export const displayName = 'aws-eks-addon';
+
+export const urls = [];
+
+export const supportsRanges = false;
+
+const versionPattern = regEx(
+  '^[vV]?(\\d+(?:\\.\\d+)*)(?<metadata>-eksbuild\\.\\d+)?$',
```

Comment:
```suggestion
  '^v?(\\d+(?:\\.\\d+)*)(?<metadata>-eksbuild\\.\\d+)?$',
```

why support upper case `V` ? This regex is also too loose. you should use `semver` module to parse instead of regex. at least add matching named capture groups like for regex versioning.

Reply:
Updated regex to capture all groups. 

And I have tried I have tried to `semver-coreced`, as this is the closest versioning.

```js
import { api as semverCoerced } from '../semver-coerced';

and did something like 

export const api: VersioningApi = {
  ...semverCoerced,
  isValid,
  isVersion,
  isCompatible,
  isGreaterThan
};

function isValid(input: string): boolean {
  if (semverCoerced.isValid(input)) {
    if (input.includes('-eksbuild.')) {
      return true;
    }
    return false;
  } else {
    return false;
  }
}

function isVersion(version: string): boolean {
  return isValid(version);
}

function isCompatible(version: string, current: string): boolean {
  return isVersion(version) && isVersion(current);
}

function isGreaterThan(version: string, other: string): boolean {
   // needs override
  return needs override;
}
```

The issue with this approach, it constantly require direct methods override. 

I have no preference, please suggest best practice and I follow.